datafusion/physical-expr-adapter/README.md

```diff
@@ -0,0 +1,14 @@
+# DataFusion Physical Expression Adapter
+
+This crate provides physical expression schema adaptation utilities for DataFusion that allow adapting a `PhysicalExpr` to different schema types.
+This handles cases such as `lit(ScalarValue::Int32(123)) = int64_column` by rewriting it to `lit(ScalarValue::Int32(123)) = cast(int64_column, 'Int32')`
+(note: this does not attempt to then simplify such expressions, that is done by shared simplifiers).
+
+## Overview
```

Comment:
I recommend putting this information on the doc comments of `PhysicalExprSchemaRewriter` itself as it will be more likely someone will find it (either in docs.rs or their IDE) than this readme

The high level overview is cool but I recommend keeping the README relatively brief